api/api.proto

```diff
@@ -5408,6 +5461,44 @@ message SdkFilesystemCheckStopRequest {
 message SdkFilesystemCheckStopResponse{
 }
 
+// SdkFilesystemCheckListSnapshotsRequest defines a request to list
+// snapshots created by fsck for a volume
+message SdkFilesystemCheckListSnapshotsRequest{
+  // Id of the volume
+  string volume_id = 1;
+}
+
+// SdkFilesystemCheckListSnapshotsResponse defines a response to list
+// snapshots created by fsck for a volume
+message SdkFilesystemCheckListSnapshotsResponse{
+  // Map of volume snapshot ids and snapshot info
+  map<string, FilesystemCheckSnapInfo> snapshots = 1;
+}
+
+// SdkFilesystemCheckDeleteSnapshotsRequest defines a request to delete
+// snapshots created by fsck for a volume
+message SdkFilesystemCheckDeleteSnapshotsRequest{
+  // Id of the volume
+  string volume_id = 1;
+}
+
+// SdkFilesystemCheckDeleteSnapshotsResponse defines a respone to delete
+// snapshots created by fsck for a volume
+message SdkFilesystemCheckDeleteSnapshotsResponse{
+}
+
+// SdkFilesystemCheckListVolumesRequest defines a request to list
+// all volumes needing fsck check/fix
+message SdkFilesystemCheckListVolumesRequest{
+}
+
+// SdkFilesystemCheckListVolumesResponse defines a response to list
+// all volumes needing fsck check/fix
+message SdkFilesystemCheckListVolumesResponse{
+  // Map of volume ids and volume info
+  map<string, FilesystemCheckVolInfo> volumes = 1;
+}
+
```

Comment:
a sample log here in the PR, as to what is the request and what is the output like will help.

Reply:
Added sample output in PR.